wiki/UI-component/Video.md

```diff
@@ -39,6 +39,147 @@ The [[Video]] component utilizes the embeddable Kaltura player <sup>[[r2](#resou
 
 <hr width="40%" />
 
+## 1. Data & content
+
+The video component intakes a Kaltura `videoId` and fetches the following data to be rendered on page as content. All of these content can be customized except the video duration.
+
+See more details at [Kaltura Player API](http://player.kaltura.com/docs/api).
+
+| Data name | Type | Allows override |
+| --------- | ---- | --------------- |
+| Name | Text | Yes |
+| Description | Text | Yes |
+| Duration | Number | No |
+| Poster image | Image | Yes |
+
+## 2. Rendering & behavior
+
+Video is used in the following components. Depending on the visual design and content requirement of the final component rendered, the above data is used differently.
+
+### Video
+
+This is the default rendering of video. It is used for embedding a video directly on page or inline.
+
+![video](https://user-images.githubusercontent.com/15144993/123469683-5cf69380-d5c1-11eb-844f-39e26678c961.png)
+
+| ID  | Data name     | Field |
+| --- | ------------- | ----- |
+| 1   | Name   | Caption |
+| 2   | Duration | appended after caption in parenthesis |
+| 3   | Description | Poster image alt tag |
+| 4   | Poster image | Image |
+
+Interactions
+
+| ID  | Element name | Event | Behavior | Notes |
+| --- | ------------ | ----- | -------- | ----- |
+| 1   | Poster image | On Click or On Enter key of link | Poster image will be replaced by Kaltura video player that will begin playing <br /> Alternative option to open in LightBoxMediaViewer. | Poster image obeys aspect ratios: 16x9, 9x16, 2x1, 1x2, 4x3, 3x4, 1x1 |
+
+### CTA - Text
+
+![CTA - Text](https://user-images.githubusercontent.com/15144993/123468111-523aff00-d5bf-11eb-8ef1-2f05c0de29c6.png)
+
+| ID  | Data name     | Field |
+| --- | ------------- | ----- |
+| 1   | Name | CTA text |
+| 2   | Duration | appended after CTA text in parenthesis |
+
+Interactions
+
+| ID  | Element name | Event | Behavior |
+| --- | ------------ | ----- | -------- |
+| 1   | CTA - Text | On Click or On Enter key of link | Open video in LightBoxMediaViewer. |
+
+
+### CTA - Button
+
+![CTA - Button](https://user-images.githubusercontent.com/15144993/123468110-523aff00-d5bf-11eb-9b1d-0c65f082d482.png)
+
+| ID  | Data name     | Field |
+| --- | ------------- | ----- |
+| 1   | Name | Button label |
+| 2   | Duration | appended after Button label in parenthesis |
+
+Interactions
```

Comment:
1. Follow the functional spec template hiearchy when possible.
2. I'd also suggest Logic as the title to align with the rest of the specs.

```suggestion
###### Logic
```